precompile/allow_list.go

```diff
@@ -0,0 +1,207 @@
+	input := make([]byte, 0, selectorLen+common.HashLength)
+	input = append(input, readAllowListSignature...)
+	input = append(input, address.Hash().Bytes()...)
+	return input
+}
+
+// writeAllowList verifies that [callerAddr] has the correct permissions to modify the allow list and if so updates [modifyAddress] permissions
+// to [role].
+func writeAllowList(evm PrecompileAccessibleState, callerAddr common.Address, modifyAddress common.Address, role AllowListRole, suppliedGas uint64, readOnly bool) (ret []byte, remainingGas uint64, err error) {
+	// Note: this should never happen since the required gas should be verified before calling Run.
+	if suppliedGas < ModifyAllowListGasCost {
+		return nil, 0, fmt.Errorf("running allow list exceeds gas allowance (%d) < (%d)", ModifyAllowListGasCost, suppliedGas)
+	}
+
+	remainingGas = suppliedGas - ModifyAllowListGasCost
+	if readOnly {
+		return nil, remainingGas, ErrReadOnlyModifyAllowList
+	}
+
+	// Verify that the caller is in the allow list and therefore has the right to modify it
+	callerStatus := GetAllowListStatus(evm.GetStateDB(), callerAddr)
+	if !callerStatus.IsAdmin() {
+		return nil, remainingGas, fmt.Errorf("%w: %s", ErrCannotModifyAllowList, callerAddr)
+	}
+
+	SetAllowListRole(evm.GetStateDB(), modifyAddress, role)
+
+	// Return an empty output and the remaining gas
+	return []byte{}, remainingGas, nil
+}
+
+// createAllowListSetter returns an execution function for setting the allow list status of the input address argument to [role].
+func createAllowListSetter(role AllowListRole) func(evm PrecompileAccessibleState, callerAddr common.Address, addr common.Address, input []byte, suppliedGas uint64, readOnly bool) (ret []byte, remainingGas uint64, err error) {
+	return func(evm PrecompileAccessibleState, callerAddr, addr common.Address, input []byte, suppliedGas uint64, readOnly bool) (ret []byte, remainingGas uint64, err error) {
+		if len(input) != common.HashLength {
+			return nil, remainingGas, fmt.Errorf("invalid input length for modifying allow list: %d", len(input))
+		}
+
+		modifyAddress := common.BytesToAddress(input)
+		return writeAllowList(evm, callerAddr, modifyAddress, role, suppliedGas, readOnly)
+	}
+}
+
+// readAllowList is the execution function for read access to the allow list.
+// parses [input] into a single address and returns the 32 byte hash that specifies the designated role of that address.
+func readAllowList(evm PrecompileAccessibleState, callerAddr common.Address, addr common.Address, input []byte, suppliedGas uint64, readOnly bool) (ret []byte, remainingGas uint64, err error) {
+	// Note: this should never happen since the required gas should be verified before calling Run.
+	if suppliedGas < ReadAllowListGasCost {
+		return nil, 0, fmt.Errorf("running allow list exceeds gas allowance (%d) < (%d)", ReadAllowListGasCost, suppliedGas)
+	}
+
+	remainingGas = suppliedGas - ReadAllowListGasCost
+
+	if len(input) != common.HashLength {
```

Comment:
address length?

Reply:
Same comment as above, this is for compatibility purposes.